lib/ace/edit_session/bracket_match.js

```diff
@@ -37,6 +37,70 @@ var Range = require("../range").Range;
 
 function BracketMatch() {
 
+
+    this.getBracketsCoor = function(position) {
+        if (position.column == 0) return null;
+
+
+        var charBeforeCursor,
+            matchLeft,
+            charAfterCursor,
+            matchRight;
+        
+        var length = -1,
+            depthLeft = 0,
+            depthRight = 0;
+
+
+        while (!matchLeft && !matchRight) {
+
+            length++;
```

Comment:
so you will always check chars on the same distance from the position? What if you have `(foo_barbar)` ? (_ stands for caret)